### DISCUSSION TITLE
Creating a layered domain?

### DISCUSSION THREAD

**Discussion by behroozkarami** [neperfepx/neper#415] [Q&A] [answered]
Dear All,

I need to create a two layered domain ( 2D)  in which the top part (subdomain) is grained and the bottom in blank. I was wondering is it possible in Neper? Any ideas are appreciated.

Thank you very much,
Behrouz

**Comment by rquey** [comment, marked as answer]
```
neper -T -dim 2 -n "from_morpho::id==1?1:100" -morpho "lamellar(w=0.9,v=(0,1,0),pos=start)::voronoi" -o d
neper -V d.tess -print img
```

![image](https://user-images.githubusercontent.com/2659376/159445628-0193b02a-0a23-4815-934e-b7d1d68ce549.png)

**Reply by behroozkarami** [reply]
Thanks a lot! That's exactly what I was looking for.
Best regards,
Behrouz

**Reply by behroozkarami** [reply]
Dear Romain,

Could you please advise me how can I apply statistical distributions option to the top layer?
I want to apply something like -morpho "diameq:lognormal(0.1,0.03),1-sphericity:lognormal(0.145,0.03)" to the grains distributed at the top layer but haven't been able to figure it out yet.

Thank you very much,
Behrouz

**Reply by rquey** [reply]
You need a multiscale cell file for that:

mymsfile:
```
1 voronoi
2 diameq:lognormal(0.1,0.03),1-sphericity:lognormal(0.145,0.03)
```

```
neper -T -dim 2 -n "from_morpho::id==1?1:100" -morpho "lamellar(w=0.9,v=(0,1,0),pos=start)::msfile(mymsfile)" -o d
neper -V d.tess -print img
```
![image](https://user-images.githubusercontent.com/2659376/161066282-4ca5bb6d-cbd0-46bd-b69d-7ff48a6590ed.png)

**Reply by behroozkarami** [reply]
Right, thank you so much.
